deploy/ControllerInit.sol

```diff
@@ -0,0 +1,171 @@
+
+import { UsdsInstance } from "lib/usds/deploy/UsdsInstance.sol";
+
+import { CCTPForwarder } from "lib/xchain-helpers/src/forwarders/CCTPForwarder.sol";
+
+import { RateLimitHelpers } from "src/RateLimitHelpers.sol";
+
+import { IALMProxy }   from "src/interfaces/IALMProxy.sol";
+import { IRateLimits } from "src/interfaces/IRateLimits.sol";
+
+import { ControllerInstance } from "./ControllerInstance.sol";
+
+interface IBufferLike {
+    function approve(address, address, uint256) external;
+}
+
+interface IMainnetControllerLike is IAccessControl {
+    function FREEZER() external view returns (bytes32);
+    function LIMIT_USDC_TO_CCTP() external view returns (bytes32);
+    function LIMIT_USDC_TO_DOMAIN() external view returns (bytes32);
+    function LIMIT_USDS_MINT() external view returns (bytes32);
+    function LIMIT_USDS_TO_USDC() external view returns (bytes32);
+    function RELAYER() external view returns (bytes32);
+}
+
+interface IForeignControllerLike is IAccessControl {
+    function FREEZER() external view returns (bytes32);
+    function LIMIT_PSM_DEPOSIT() external view returns (bytes32);
+    function LIMIT_PSM_WITHDRAW() external view returns (bytes32);
+    function RELAYER() external view returns (bytes32);
+}
+
+interface IPSMLike {
+    function kiss(address) external;
+}
+
+interface IVaultLike {
+    function rely(address) external;
+}
+
+library MainnetControllerInit {
+
+    struct RateLimitData {
+        uint256 maxAmount;
+        uint256 slope;
+    }
+
+    function subDaoInit(
+        address              freezer,
+        address              relayer,
+        ControllerInstance   memory controllerInst,
+        AllocatorIlkInstance memory ilkInst,
+        UsdsInstance         memory usdsInst,
+        RateLimitData        memory usdsMintData,
+        RateLimitData        memory usdcToUsdsData,
+        RateLimitData        memory usdcToCctpData,
+        RateLimitData        memory cctpToBaseDomainData
+    )
+        internal
+    {
+        IALMProxy   almProxy   = IALMProxy(controllerInst.almProxy);
+        IRateLimits rateLimits = IRateLimits(controllerInst.rateLimits);
+
+        IMainnetControllerLike controller = IMainnetControllerLike(controllerInst.controller);
+
+        // Step 1: Configure ACL permissions for controller and almProxy
+
+        controller.grantRole(controller.FREEZER(), freezer);
+        controller.grantRole(controller.RELAYER(), relayer);
+
+        almProxy.grantRole(almProxy.CONTROLLER(), address(controller));
+
+        rateLimits.grantRole(rateLimits.CONTROLLER(), address(controller));
+
+        // Step 2: Configure all rate limits for controller, using Base as only domain
+
+        bytes32 domainKeyBase = RateLimitHelpers.makeDomainKey(
+            controller.LIMIT_USDC_TO_DOMAIN(),
+            CCTPForwarder.DOMAIN_ID_CIRCLE_BASE
+        );
+
+        rateLimits.setRateLimitData(controller.LIMIT_USDS_MINT(),    usdsMintData.maxAmount,         usdsMintData.slope);
```

Comment:
Should be an easy way to specify unlimited limit which should in turn use the `setUnlimited...`. Alternatively we can just infer unlimited by checking the `maxAmount == type(uint256).max` and if so ensure the slope is 0.

Reply:
Added, need to add tests

Reply:
Added tests